### PR TITLE
M5327 Global localized exception handling review changes

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits

### CODE REVIEW

File: molgenis-data-i18n/src/test/java/org/molgenis/data/i18n/LanguageRepositoryDecoratorTest.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void testAddExistingLanguage()
 		verify(delegateRepository).add(language);
 	}
 
-	@Test(expectedExceptions = AddLanguageException.class)
+	@Test(expectedExceptions = RepositoryCapabilityException.class, expectedExceptionsMessageRegExp = "repository:Language capability:WRITABLE")
```
Comment (fdlk) [inline molgenis-data-i18n/src/test/java/org/molgenis/data/i18n/LanguageRepositoryDecoratorTest.java]: Filter the WRITABLE capability from the capabilities listed by getCapabilities()

File: molgenis-gavin/src/main/java/org/molgenis/gavin/job/GavinJobFactory.java [hunk 2]
```diff
@@ -88,18 +88,16 @@ public GavinJob createJob(GavinJobExecution gavinJobExecution)
...
 	@RunAsSystem
-	public GavinJobExecution findGavinJobExecution(String jobIdentifier) throws JobNotFoundException
+	public GavinJobExecution findGavinJobExecution(String jobIdentifier)
 	{
 		GavinJobExecution result = dataService.findOneById(GAVIN_JOB_EXECUTION, jobIdentifier, GavinJobExecution.class);
 		if (result == null)
 		{
-
-			throw new JobNotFoundException();
+			throw new UnknownEntityException(dataService.getEntityType(GAVIN_JOB_EXECUTION), jobIdentifier);
```
Comment (fdlk) [inline molgenis-gavin/src/main/java/org/molgenis/gavin/job/GavinJobFactory.java]: autowire the metadata

File: molgenis-data-i18n/src/test/java/org/molgenis/data/i18n/LanguageRepositoryDecoratorTest.java (hunk 1 above)
Comment (dennishendriksen) [reply molgenis-data-i18n/src/test/java/org/molgenis/data/i18n/LanguageRepositoryDecoratorTest.java]: I don't understand what you mean with this ...

Comment (fdlk) [reply molgenis-data-i18n/src/test/java/org/molgenis/data/i18n/LanguageRepositoryDecoratorTest.java]: I meant that if the repository is going to throw an exception that says you cannot write to it, it should not claim that you can write to it when asked about its capabilities. But as discussed, the repository is "somewhat writable" :)

**Pull request merged**